### DISCUSSION TITLE
When login into docker registry using docker login command , 2 requests get sent and first one fails with error error "authorizing context: basic authentication challenge for realm \\\"docker-registry-realm\\\": invalid authorization credential\"

### DISCUSSION THREAD

**Discussion by rahulbapumore** [distribution/distribution#4450] [Q&A]
We notices strange behavior with docker distribution version 2.8.2
For each docker login command we run to that registry, inside logs we see below sets of logs out which one is invalid credentials error and other one is authorized completed leading to succeeded login.
We are able to login successfully but inside logs error is getting logged.

Command we are running is - 

```bash
docker  login private-registry-url:80
Username: admin
Password:
WARNING! Your password will be stored unencrypted in /root/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded
```

And below are the logs logged for docker registry -

```
{"message":"**error authorizing context: basic authentication challenge for realm  \\\"docker-registry-realm\\\": invalid authorization credential\**","metadata":{"container_name":"registry","namespace":"namespace1","pod_name":"distribution-7c79644d8b-ndc97"},"service_id":"distribution","severity":"warning","timestamp":"2024-08-28T15:42:48.932+00:00","version":"1.2.0"}
{"message":"fc00:1000::a98e - - [28/Aug/2024:15:42:48 +0000] \"GET /v2/ HTTP/1.1\" 401 87 \"\" \"docker/20.10.7 go/go1.13.15 git-commit/b0f5bc3 kernel/4.15.0-118-generic os/linux arch/amd64 UpstreamClient(Docker-Client/20.10.7 \\\\(linux\\\\))\"","metadata":{"container_name":"registry","namespace":"namespace1","pod_name":"distribution-7c79644d8b-ndc97"},"service_id":"distribution","severity":"info","timestamp":"2024-08-28T15:42:48.932+00:00","version":"1.2.0"}
{"message":"authorizing request","metadata":{"container_name":"registry","namespace":"namespace1","pod_name":"distribution-7c79644d8b-ndc97"},"service_id":"distribution","severity":"debug","timestamp":"2024-08-28T15:42:48.946+00:00","version":"1.2.0"}
{"message":"authorized request","metadata":{"container_name":"registry","namespace":"namespace1","pod_name":"distribution-7c79644d8b-ndc97"},"service_id":"distribution","severity":"info","timestamp":"2024-08-28T15:42:48.949+00:00","version":"1.2.0"}
{"message":"response completed","metadata":{"container_name":"registry","namespace":"namespace1","pod_name":"distribution-7c79644d8b-ndc97"},"service_id":"distribution","severity":"info","timestamp":"2024-08-28T15:42:48.949+00:00","version":"1.2.0"}
{"message":"192.168.169.153 - - [28/Aug/2024:15:42:48 +0000] \"GET /v2/ HTTP/1.1\" 200 2 \"\" \"docker/20.10.7 go/go1.13.15 git-commit/b0f5bc3 kernel/4.15.0-118-generic os/linux arch/amd64 UpstreamClient(Docker-Client/20.10.7 \\\\(linux\\\\))\"","metadata":{"container_name":"registry","namespace":"namespace1","pod_name":"distribution-7c79644d8b-ndc97"},"service_id":"distribution","severity":"info","timestamp":"2024-08-28T15:42:48.949+00:00","version":"1.2.0"}
```

Why below line is printed in logs?
**error authorizing context: basic authentication challenge for realm  \\\"docker-registry-realm\\\": invalid authorization credential\**

**Comment by thaJeztah** [comment]
If memory serves me well, the first request to the `/v2/` endpoint is without authentication, and used for discovery; see https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#determining-support

In that case, as outlined (e.g.) in https://github.com/distribution/distribution/blob/main/docs/content/spec/auth/token.md, the client calls the endpoint; the registry will respond whether (or not) authentication is needed (and what authentication mechanism to use), which is used by the client to do actual authentication.

**Reply by milosgajdos** [reply]
This is correct indeed. Also, @rahulbapumore I'd strongly recommend upgrading your distribution deployment. `2.8.2` is ancient. Literally. Upgrading will not remove what you're saying due to what Seb says, but that version has more bugs than the sidekick utility I wrote this morning for cleaning some data on my laptop. So, consider that.

**Comment by rahulbapumore** [comment]
Okay thanks for clarification..
closing this query
